control/controlclient/auto.go

```diff
@@ -561,6 +561,17 @@ func (c *Auto) SetNetInfo(ni *tailcfg.NetInfo) {
 	c.sendNewMapRequest()
 }
 
+// SetTKAHead updates the TKA head hash that map-request infrastructure sends.
+// If the value has changed, a new map-request is queued.
+func (c *Auto) SetTKAHead(headHash string) {
+	if !c.direct.SetTKAHead(headHash) {
+		return
+	}
+
+	// Send new head hash to server
+	c.sendNewMapRequest()
```

Comment:
we're trying to get away from this, restart map requests on changes. You send a new discrete HTTP request instead about what you want to say.

Reply:
We need this information in computing the map response for disablement.

Are there other places we tx information to control in a discrete RPC in the manner you're describing?

Reply:
Existing ones are:

* the "lite endpoint updates" which are map requests in name only but are effectively separate RPCs (and don't break the map poll)
* set-dns
* update-health

Reply:
Closing this out: The sync-send RPC (which has to be sent anyway) carries this information and on the control site gets wired into the node wakeup mechanics, hence avoiding the need to restart the map poll.